subsys/net/l2/openthread/openthread.c

```diff
@@ -302,15 +338,22 @@ static int openthread_init(struct net_if *iface)
 	NET_INFO("OpenThread version: %s",
 		    otGetVersionString());
 
+#if !defined(CONFIG_OPENTHREAD_JOINER_AUTOSTART)
 	otThreadSetNetworkName(ot_context->instance, CONFIG_OPENTHREAD_NETWORK_NAME);
 	NET_INFO("Network name:   %s",
 		 log_strdup(otThreadGetNetworkName(ot_context->instance)));
 
 	otLinkSetChannel(ot_context->instance, CONFIG_OPENTHREAD_CHANNEL);
 	otLinkSetPanId(ot_context->instance, CONFIG_OPENTHREAD_PANID);
 	otThreadSetExtendedPanId(ot_context->instance, &xpanid);
+#endif
+
 	otIp6SetEnabled(ot_context->instance, true);
+
+#if !defined(CONFIG_OPENTHREAD_JOINER_AUTOSTART)
 	otThreadSetEnabled(ot_context->instance, true);
```

Comment:
Please move this below, it'll be easier to follow how network starts if we keep the logic in on place. I mean something like this:
```
#if !defined(CONFIG_OPENTHREAD_JOINER_AUTOSTART)
    otThreadSetEnabled(ot_context->instance, true);
#else
    openthread_joiner_autostart(iface);
#endif
```

Reply:
Thank you! I fixed it.